lib/avirwrapper.php

```diff
@@ -136,4 +144,21 @@ private function isWritingMode($mode){
 		);
 		return in_array($cleanMode, $this->writingModes);
 	}
+
+	/**
+	 * checks the size for webdav PUT requests. defaults to true
+	 * @param $filename
+	 * @return bool
+	 */
+	private function isScannableSize($filename) {
+		$scanSizeLimit = intval($this->appConfig->getAvMaxFileSize());
+		$size = false;
+
+		// PUT via webdav
+		if ($_SERVER['REQUEST_METHOD'] === 'PUT' && isset($_SERVER['CONTENT_LENGTH'])) {
```

Comment:
Yes, the $_SERVER variable can be set during unit testing and private methods can be made public.